transmission_interface/include/transmission_interface/simple_transmission.h

```diff
@@ -0,0 +1,211 @@
+     */
+    void joint_to_actuator() override;
+
+    std::size_t num_actuators() const override {return 1;}
+    std::size_t num_joints()    const override {return 1;}
+
+    double get_actuator_reduction() const {return reduction_;}
+    double get_joint_offset()       const {return jnt_offset_;}
+
+protected:
+    double reduction_;
+    double jnt_offset_;
+
+    JointHandle joint_position = {"", "", nullptr};
+    JointHandle joint_velocity = {"", "", nullptr};
+    JointHandle joint_effort = {"", "", nullptr};
+
+    ActuatorHandle actuator_position = {"", "", nullptr};
+    ActuatorHandle actuator_velocity = {"", "", nullptr};
+    ActuatorHandle actuator_effort = {"", "", nullptr};
+  };
+
+  inline SimpleTransmission::SimpleTransmission(
+    const double reduction,
+    const double joint_offset)
+  : reduction_(reduction),
+    jnt_offset_(joint_offset)
+  {
+    if (0.0 == reduction_) {
+      throw Exception("Transmission reduction ratio cannot be zero.");
+    }
+  }
+
+  template<class HandleType>
+  HandleType get_by_interface(
+    const std::vector<HandleType> & handles,
+    const std::string & interface_name)
+  {
+    const auto result =
+      std::find_if(
+      handles.cbegin(), handles.cend(), [&interface_name](
+        const auto handle) {return handle.get_interface_name() == interface_name;});
+    if (result == handles.cend()) {
+      return HandleType("", "", nullptr);
+    }
+    return *result;
+  }
+
+  void SimpleTransmission::configure(
+    const std::vector<JointHandle> & joint_handles,
+    const std::vector<ActuatorHandle> & actuator_handles)
+  {
+    // check that joint / act names are identical
+    // check that joint interfaces and actuator interfaces match
+
+    joint_position = get_by_interface(joint_handles, "position");
+    joint_velocity = get_by_interface(joint_handles, "velocity");
+    joint_effort = get_by_interface(joint_handles, "effort");
+
+    actuator_position = get_by_interface(actuator_handles, "position");
+    actuator_velocity = get_by_interface(actuator_handles, "velocity");
+    actuator_effort = get_by_interface(actuator_handles, "effort");
+  }
+
+  inline void SimpleTransmission::actuator_to_joint()
+  {
+    if (joint_effort && actuator_effort) {
+      joint_effort.set_value(actuator_effort.get_value() * reduction_);
+    }
+
+    if (joint_velocity && actuator_velocity) {
+      joint_velocity.set_value(actuator_velocity.get_value() / reduction_);
+    }
+
+    if (joint_position && actuator_position) {
+      joint_position.set_value(actuator_position.get_value() / reduction_ + jnt_offset_);
+    }
+
+//   *jnt_data.absolute_position[0] = *act_data.absolute_position[0] / reduction_ + jnt_offset_;
+//   *jnt_data.torque_sensor[0] = *act_data.torque_sensor[0] * reduction_;
+  }
+
+
+  inline void SimpleTransmission::joint_to_actuator()
+  {
+    if (joint_effort && actuator_effort) {
+      actuator_effort.set_value(joint_effort.get_value() / reduction_);
```

Comment:
I know this was present in the original implementation, but I'm wondering in which cases should the reduction be applied to the effort interface. E.g., it does not make sense if you are using current as effort.

Reply:
What should we do here instead? Acceleration? Torque? same handling for both?